hastexo/jobs.py

```diff
@@ -90,37 +90,43 @@ def suspend_stack(self, stack):
         Suspend the stack.
 
         """
-        credentials = get_credentials(self.settings, stack.provider)
-        heat_client = self.get_heat_client(credentials)
-
         try:
-            heat_stack = heat_client.stacks.get(stack_id=stack.name)
-        except HTTPNotFound:
-            heat_status = DELETED_STATE
-        else:
-            heat_status = heat_stack.stack_status
+            credentials = get_credentials(self.settings, stack.provider)
+            heat_client = self.get_heat_client(credentials)
 
-        if heat_status in UP_STATES:
-            self.log("Suspending stack [%s]." % stack.name)
+            try:
+                heat_stack = heat_client.stacks.get(stack_id=stack.name)
+            except HTTPNotFound:
+                heat_status = DELETED_STATE
+            else:
+                heat_status = heat_stack.stack_status
 
-            # Suspend stack
-            heat_client.actions.suspend(stack_id=stack.name)
+            if heat_status in UP_STATES:
+                self.log("Suspending stack [%s]." % stack.name)
```

Comment:
Stack names get reused over time when learners reset labs. When examining this log message in the course of troubleshooting, it's obviously possible to use just the stack name in combination with `openstack stack list --deleted | grep $name` to then retrieve the uuid log of the deleted stack, but perhaps we want to instead log both the name and the uuid here?

Reply:
Addresse don fix-up 4c682b4

Reply:
Looks good to me; please squash — thanks!